.idx-templates/latest/dev.nix

```diff
@@ -1,22 +1,29 @@
 # To learn more about how to use Nix to configure your environment
 # see: https://developers.google.com/idx/guides/customize-idx-env
-{ pkgs, ... }:
-{
+{ pkgs, ... }: {
   # Which nixpkgs channel to use.
   channel = "stable-24.05"; # or "unstable"
   # Use https://search.nixos.org/packages to find packages
   packages = [ pkgs.nodejs_20 ];
   # Sets environment variables in the workspace
   env = { };
-  idx = {
+  idx = let
     # Search for the extensions you want on https://open-vsx.org/ and use "publisher.id"
-    extensions = [
-      "astro-build.astro-vscode"
-    ];
+    # Workaround for https://community.idx.dev/t/8374
+    # `idx.extensions` only work for extensions declared in the Nixpkgs derivation used internally by IDX,
+    # forked at some point from the official `release-24.05` channel.
+    # Instead of relying on IDX internal installation, we also add a startup script to install/update the extensions.
+    extensions = [ "astro-build.astro-vscode" ];
+  in {
+    inherit extensions;
     workspace = {
       # Runs when a workspace is first created with this `dev.nix` file
       onCreate = {
-        install = "npm ci --prefer-offline --no-audit --no-progress --timing || npm i --no-audit --no-progress --timing";
+        install-deps =
+          "npm ci --prefer-offline --no-audit --no-progress --timing || npm i --no-audit --no-progress --timing";
```

Comment:
This is causing an error `onCreate` 
```sh
The `npm ci` command can only install with an existing package-lock.json or
npm error npm-shrinkwrap.json with lockfileVersion >= 1. Run an install with npm@5 or
npm error later to generate a package-lock.json file, then try again.
```
An easy remedy would be to either include the `package-lock.json` or have these `npm ci` commands be `npm install` first to allow the generation of the lock file. 
```sh
 install = "npm install --no-audit --no-progress --timing";
 install-deps = "npm install --no-audit --no-progress --timing";
```

Reply:
This is intentional. The `npm install` is added as an alternative at the end (`|| npm i ...`).

If the project has a package-lock it will be used with a clean-install, if it doesn't the clean-install will error and the normal intall will run.

Reply:
I did see that, however I felt that the requirement for a clean install over a standard install since a clean install does require a lock file to do its thang, was a bit elaborate since it would most likely trigger the second condition as it is, 
sorry to nit over this one, I will resolve and defer to your better judgement @Fryuni